src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -1,37 +1,321 @@
-import {LightningElement, api, track} from 'lwc';
-import GeFormService from 'c/geFormService';
+import { LightningElement, api, track } from 'lwc';
+import { NavigationMixin } from 'lightning/navigation';
+import saveAndDryRunDataImport from '@salesforce/apex/GE_GiftEntryController.saveAndDryRunDataImport';
+import createTransaction from '@salesforce/apex/GE_GiftEntryController.createTransaction';
+import upsertDataImport from '@salesforce/apex/GE_GiftEntryController.upsertDataImport';
+import processDataImport from '@salesforce/apex/GE_GiftEntryController.processDataImport';
+
+//import GeFormService from 'c/geFormService';
+import { showToast } from 'c/utilTemplateBuilder';
+import { deepClone } from 'c/utilCommon';
+import { registerListener, unregisterListener } from 'c/pubsubNoPageRef';
+import GeLabelService from 'c/geLabelService';
+
 import DATA_IMPORT_BATCH_OBJECT from '@salesforce/schema/DataImportBatch__c';
+import DI_PAYMENT_AUTHORIZE_TOKEN_FIELD from '@salesforce/schema/DataImport__c.Payment_Authorization_Token__c';
+import DI_PAYMENT_STATUS_FIELD from '@salesforce/schema/DataImport__c.Payment_Status__c';
+import DI_PAYMENT_DECLINED_REASON_FIELD from '@salesforce/schema/DataImport__c.Payment_Declined_Reason__c';
+import DI_PAYMENT_METHOD_FIELD from '@salesforce/schema/DataImport__c.Payment_Method__c';
+
+const PAYMENT_STATUS__C = DI_PAYMENT_STATUS_FIELD.fieldApiName;
+const PAYMENT_DECLINED_REASON__C = DI_PAYMENT_DECLINED_REASON_FIELD.fieldApiName;
+const PAYMENT_AUTHORIZE_TOKEN__C = DI_PAYMENT_AUTHORIZE_TOKEN_FIELD.fieldApiName;
+const PAYMENT_METHOD__C = DI_PAYMENT_METHOD_FIELD.fieldApiName;
+const CAPTURED = 'CAPTURED';
+const TOKENIZE_TIMEOUT = 10000; // 10 seconds, long enough for cold starts?
+const PAYMENT_TRANSACTION_STATUS_ENUM = Object.freeze({
+    PENDING: 'PENDING',
+    AUTHORIZED: 'AUTHORIZED',
+    CANCELED: 'CANCELED',
+    CAPTURED: 'CAPTURED',
+    DECLINED: 'DECLINED',
+    NONRETRYABLEERROR: 'NONRETRYABLEERROR',
+    RETRYABLEERROR: 'RETRYABLEERROR',
+    REFUNDISSUED: 'REFUNDISSUED'
+});
 
-export default class GeGiftEntryFormApp extends LightningElement {
+export default class GeGiftEntryFormApp extends NavigationMixin(LightningElement) {
     @api recordId;
     @api sObjectName;
 
     @track isPermissionError;
+    @track loadingText = 'Saving data import...';
```

Comment:
Could we use a custom label here?

Reply:
@vmartinov-sforg - updated with custom labels 👍

Reply:
@psadie-sfdo should the 'Saving data import...' text still be here if we are using custom labels?

Reply:
It sure should be. Updating for the next commit.